### DISCUSSION TITLE
Update variable with wire:poll

### DISCUSSION THREAD

**Discussion by tomickigrzegorz** [livewire/livewire#4921] [1. Help]
I have a variable in blade.php that is dynamically updated.

```html
<script wire:poll>
   let $layerFlights = @json($flights);
</script>
```
From an external js script which I add to blade.php `<script src="./js/script.js"></script>`

In the script, I have the usual reading of a variable.

```js
setInterval(() => {
  console.log($layerFlights);
}, 2000);
```
The affected variable follows, I can see in the DOM but I can't get it from an external js script.

How can you get to this variable after the data update?

**Comment by joshhanley** [comment]
@tomik23 script tags aren't re-evaluated, so changing the value won't do anything. If you need to get the latest value from Livewire, you should use Livewire's javascript object and access properties from the component directly.

Using Alpine for this is even easier using `@entangle`.

Hope this helps!

**Reply by tomickigrzegorz** [reply]
@joshhanley Thanks, it helped a bit, but I'm not sure if I understood correctly ;)
Maybe I will mark that I am new to laravel and livewire so be patient ;)

I have a component in, let's say, this format:
```html
<div>
    <script wire:poll>
        let $layerFlights = @json($flights);
    </script>
    <scirpt scr="./script.js"></script>
</div>
```
The **$layerFlights** variable changes every 2s. 
The variable updates because I can see the DOM changing in devTools.
I have `setInterval` set to 2s in the script where I'm doing the usual `console.log($layerFlights)`

Now I will refer to "you should use Livewire's javascript object and access properties from the component directly"

So in the blade component I need to get to the variable in my script?
But how? Maybe you have some such example or you can guide me in the documentation. Unfortunately I don't use Alpine so I can't use @entangle.

Thanks for any help.

**Reply by joshhanley** [reply]
@tomik23 no worries!

So although the code for layerFlights is updated in the DOM, the updated javascript code is not evaluated, so if you interact with layerFlights, it will have the original data.

The best way to access the data (without Alpine), is to save the data to a property on the Livewire component.

So if you created a property on your Livewire component called $flights and added the data to it, then you could access in your javascript using `@this.flights`

Like this

```blade
<div>
    <script wire:poll>
        console.log(@this.flights)
    </script>
    <scirpt scr="./script.js"></script>
</div>
```

It all depends on what the flight data is, if it's an array it will work as is, but if it's an eloquent collection of models, you will need to whitelist any properties that you want the front end to be able to access. See this section about adding rules for binding (it's the same process to make data available to the front end) https://laravel-livewire.com/docs/2.x/properties#binding-models

Hope this helps!
